pkg/machinery/config/types/v1alpha1/generate/generate.go

```diff
@@ -208,41 +208,26 @@ func NewSecretsBundle(clock Clock, opts ...GenOption) (*SecretsBundle, error) {
 		return options.Secrets, nil
 	}
 
+	return NewSecretsBundleWithOptions(clock, options.VersionContract, "", "")
+}
+
+// NewSecretsBundleWithOptions creates secrets bundle,
+// generating all secrets or reading from the input options if provided.
+//
+//nolint:gocyclo
+func NewSecretsBundleWithOptions(clock Clock, versionContract *config.VersionContract, pkiDir, bootstrapToken string) (*SecretsBundle, error) {
```

Comment:
from the API perspective it feels like it should be a different method.

we have `NewSecretsBundle` which gives us all fresh unique PKI
we can have `NewSecretsBundleFromKubernetesPKI` which imports whatever it can from Kubernetes PKI and fills in the missing stuff. 

The implementation might be that we have a method which fills in missing secrets (generates them), and the `NewSecretsBundle` just calls it, while `NewSecretsBundleFromKubernetesPKI` first fills in what it can, and that calls that common shared method?

Right now it feels a bit of a hard to follow code, and this code is very critical to be reviewable

Reply:
I did it the way you suggested, you can have another look.